### PR TITLE
feat: Show contextual search menu for debugger errors

### LINKED ISSUE #7149: [Feature] Show contextual search menu in the debugger
## Summary

https://www.figma.com/file/EdC0j0XwxyC9Flqzxyoxuf/Debugger?node-id=2677%3A3923

Show contextual(based on error type) search menu in the debugger

VALIDATION - Copy / Search Docs / Snippet/ Ask Intercom
PARSE - Copy / Search Google/Snippet
LINT - Copy / Search Google/Snippet
PLUGIN_EXECUTION
&nbsp;&nbsp;DATASOURCE_CONFIGURATION_ERROR - Copy / Search Google / Ask Intercom
&nbsp;&nbsp;PLUGIN_ERROR - Copy / Search Google / Ask Intercom
&nbsp;&nbsp;CONNECTIVITY_ERROR - Copy / Search Docs
&nbsp;&nbsp;ACTION_CONFIGURATION_ERROR - Copy / Search Docs / Ask Intercom

### PR DESCRIPTION
**Pull request by akash-codemonk**
Show a menu on click of debugger errors with search options. 

Fixes #7149 

## Type of change

- New feature (non-breaking change which adds functionality)

## How Has This Been Tested?

Cypress test

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

## Test coverage results :test_tube:
<details><summary>:red_circle: Total coverage has decreased</summary>

    // Code coverage diff between base branch:release and head branch: feat/debugger-search-menu 
Status | File | % Stmts | % Branch | % Funcs | % Lines 
 -----|-----|---------|----------|---------|------ 
 :red_circle: | total | 55.24 **(0.01)** | 37.21 **(0)** | 34.24 **(-0.02)** | 55.79 **(0)**
 :red_circle: | app/client/src/components/ads/Icon.tsx | 55.12 **(-0.44)** | 37.8 **(-0.6)** | 100 **(0)** | 54.97 **(-0.44)**
 :red_circle: | app/client/src/components/editorComponents/CodeEditor/EvaluatedValuePopup.tsx | 66.1 **(-0.29)** | 54.05 **(-1.21)** | 39.39 **(0)** | 66.67 **(-0.32)**
 :sparkles: :new: | **app/client/src/components/editorComponents/Debugger/ContextualMenu.tsx** | **47.14** | **21.62** | **12.5** | **47.14**
 :green_circle: | app/client/src/components/editorComponents/Debugger/DebugCTA.tsx | 55.56 **(1.01)** | 60 **(26.67)** | 0 **(0)** | 57.14 **(0)**
 :green_circle: | app/client/src/components/editorComponents/Debugger/LogItem.tsx | 37.29 **(1.39)** | 25 **(8.33)** | 0 **(0)** | 38.18 **(2.18)**
 :green_circle: | app/client/src/constants/messages.ts | 71.35 **(0.48)** | 100 **(0)** | 13.58 **(1.46)** | 78.87 **(0.26)**
 :red_circle: | app/client/src/utils/autocomplete/TernServer.ts | 50.47 **(-0.24)** | 38.26 **(-0.87)** | 36.21 **(0)** | 55.08 **(-0.27)**</details>

### CODE REVIEW

File: app/client/src/components/ads/MenuItem.tsx [hunk 1]
```diff
@@ -110,7 +112,7 @@ const MenuItemContent = forwardRef(
         selected={props.selected}
         type={props.type}
       >
-        <IconContainer className={props.className}>
+        <IconContainer className={props.containerClassName}>
```
Comment (rishabhsaxena) [inline app/client/src/components/ads/MenuItem.tsx:115]: should we keep this as `className={props.className}` and use another prop for the `ItemRow`, so we can be sure existing components are not affected? Can ignore this if that's already handled

File: app/client/src/components/editorComponents/Debugger/DebugCTA.tsx [hunk 2]
```diff
@@ -7,6 +7,71 @@ import { Classes, Variant } from "components/ads/common";
...
+            offset: "0, 0",
+          },
+        }}
+        position={Position.BOTTOM_RIGHT}
+      >
+        <EVDebugButton>
+          DEBUG
+          <Icon
+            fillColor={"#F22B2B"}
+            hoverFillColor={"#F22B2B"}
```
Comment (rishabhsaxena) [inline app/client/src/components/editorComponents/Debugger/DebugCTA.tsx:66]: nit: should use color constants

File: app/client/src/components/editorComponents/Debugger/ContextualMenu.tsx [hunk 3]
```diff
@@ -0,0 +1,268 @@
...
+    CONTEXT_MENU_ACTIONS.DOCS,
+    CONTEXT_MENU_ACTIONS.SNIPPET,
+    CONTEXT_MENU_ACTIONS.GOOGLE,
+    CONTEXT_MENU_ACTIONS.INTERCOM,
+  ];
+
+  if (subType) {
+    switch (subType) {
+      // These types are sent by the server
+      case "DATASOURCE_CONFIGURATION_ERROR":
```
Comment (rishabhsaxena) [inline app/client/src/components/editorComponents/Debugger/ContextualMenu.tsx:56]: nit: should define enums for these

File: app/client/src/components/editorComponents/Debugger/ContextualMenu.tsx [hunk 4]
```diff
@@ -0,0 +1,268 @@
...
+        return [
+          CONTEXT_MENU_ACTIONS.COPY,
+          CONTEXT_MENU_ACTIONS.GOOGLE,
+          CONTEXT_MENU_ACTIONS.INTERCOM,
```
Comment (rishabhsaxena) [inline app/client/src/components/editorComponents/Debugger/ContextualMenu.tsx:60]: nit: we could hide the option to ping on intercom if it's disabled

**Review by rishabhsaxena — APPROVED** [review]

File: app/client/src/components/ads/MenuItem.tsx (hunk 1 above)
Comment (akash-codemonk) [reply app/client/src/components/ads/MenuItem.tsx:115]: Have updated this where it is being used.

**Review by rishabhsaxena — APPROVED** [review]
nitpick: should we dismiss the menu when an option is clicked?

**Comment by akash-codemonk** [comment]
> nitpick: should we dismiss the menu when an option is clicked?

Yeah, makes sense right. Otherwise it would just stay there.

**Comment by rishabhsaxena** [comment]
> > nitpick: should we dismiss the menu when an option is clicked?
> 
> Yeah, makes sense right. Otherwise it would just stay there.

Yeah my bad, I had thought it was being persisted still

**Pull request merged**